### ISSUE TITLE
Strange Results on first step

### ISSUE THREAD

**Issue by nbardy** [lucidrains/lion-pytorch#24]
I'm finetuning SD 1.5 at high 576 batch size.(24 gradient accumulation right now for testing on 1 GPU before scaling up)

Trying to get LION working. Getting very strange results on the first step. Seems to reset the model to a weird texture filled state

Step 0 on the left and Step 1 on the right

<img width="1169" alt="image" src="https://user-images.githubusercontent.com/1278972/231574305-2c97334a-9bbc-4c8f-80d1-0395ba899b6a.png">

Here is one I let run longer, It seems to actually be converging 🤔 , but still has the same reset problem at the start.
Step 500, Step 1000, Step 1500
<img width="1168" alt="image" src="https://user-images.githubusercontent.com/1278972/231574478-f8ac64a2-99a1-4087-9bd2-8f069a51cf4f.png">

Relevant Code:
```
       from lion_pytorch import Lion

        optimizer  = Lion(
            params_to_optimize,
            lr=args.learning_rate,
            weight_decay=1e-2,
            betas=(0.95,0.98),
            use_triton=True # set this to True to use cuda kernel w/ Triton lang (Tillet et al)
        )
   ```
   
Relevant parameters:
```
--batch_size=24 --learning_rate=7.0e-7 --gradient_accumulation_steps=24 --lion_opt --lr_end=7.0e-9 --lr_scheduler=cosine_with_restarts --lr_num_cycles=20 --lr_warmup_steps=0  --max_train_steps=10000 --mixed_precision bf16
```

No apparent sharp decrease in loss?
<img width="1189" alt="image" src="https://user-images.githubusercontent.com/1278972/231586689-ba500df5-78f5-4b49-b3db-f0663c6aad74.png">

More samples across many prompts:

<img width="1170" alt="image" src="https://user-images.githubusercontent.com/1278972/231586524-d2374b3a-4c48-4383-b7cf-a61e63cebf2f.png">

Note black squares are just NSFW filter I believe

**Comment by xiangning-chen** [comment]
Hi, thanks for the datapoint.

Do you have a comparison of the commands used for running with Lion and AdamW?

**Comment by nbardy** [comment]
@xiangning-chen same command besides the lr_opt value

**Comment by xiangning-chen** [comment]
Oh I meant the learning_rate, lr_end, and weight decay comparison for Lion and AdamW.

**Comment by nbardy** [comment]
They are in the main post.

‘Relevant Code’ is for lion 
‘Relevant parameters’ is for Adam

**Comment by xiangning-chen** [comment]
@nbardy Sorry I'm a bit confused, in `Relevant parameters` you set the `--lion_opt` flag, but this is for Adam?
Can you please just tell me the **learning_rate**, **lr_end**, and **weight decay** for Lion and AdamW respectively, thanks!

**Comment by mitchellnw** [comment]
@nbardy do you get this behaviour without triton?

one thing i noticed here is that the triton code uses auto-tune + in-place updates which may cause issues. on the first step multiple differnt kernels will be launched which all do the same thing to see what is fastest. this is unique to the first step. usually this is not a problem when training from scratch as warmup is used but it may be here

**Comment by lucidrains** [comment]
@mitchellnw thanks for bringing this to my attention Mitchell!

@nbardy do you want to see if https://github.com/lucidrains/lion-pytorch/commit/6ab873a380b47ebc5ea6f68ea588606daebb8b85 addresses the issue?

**Comment by nbardy** [comment]
I have finally got back to training more diffusion models.

Tried upgrading to `lion-pytorch==0.1.2` and still getting a reset it seems on first step


https://wandb.ai/nbardy-facet/sd_xl_train_t2iadapter/runs/eey3bj1n?workspace=user-nbardy-facet

**Comment by nbardy** [comment]
```
lion-pytorch==0.1.2
pytorch-triton==2.1.0+e650d3708b
triton==2.0.0
torch==2.0.1

```

**Comment by nbardy** [comment]
Turned off lion and it’s still there. This is probably something else from my changes. Will test more next week.

**Comment by nbardy** [comment]
Confirmed fixed.
